File: ios/modules/Eddystone.swift
import Foundation

struct Eddystone: Beacon {
    let type: String = "eddystone"
    let uuid: String
    let name: String?
    let namespace: String
    let instance: String
    let rssi: Int
    let txPower: Int
    let adHex: String

    func toMap() -> [String: Any?] {
        return [
            "type": type,
            "uuid": uuid,
            "name": name,
            "namespace": namespace,
            "instance": instance,
            "rssi": rssi,
            "txPower": txPower,
            "adHex": adHex
        ]
    }
}

File: android/src/main/java/expo/modules/beacons/modules/EddyStone.kt
package expo.modules.beacons.modules

data class EddyStone(
    override val uuid: String,
    val address: String,
    val name: String?,
    val namespace: String,
    val instance: String,
    val rssi: Int,
    val txPower: Int,
    val adHex: String,
) : Beacon {
    override fun toMap(): Map<String, Any?> = mapOf(
        "type" to "eddystone",
        "address" to address,
        "uuid" to uuid,
        "name" to name,
        "namespace" to namespace,
        "instance" to instance,
        "rssi" to rssi,
        "txPower" to txPower,
        "adHex" to adHex,
    )
}
